tests/integration-tests/tests/common/utils.py

```diff
@@ -108,17 +109,21 @@ def retrieve_pcluster_ami_without_standard_naming(region, os, version, architect
                 Filters=[
                     {"Name": "name", "Values": [official_ami_name]},
                     {"Name": "architecture", "Values": [architecture]},
+                    {"Name": "is-public", "Values": ["true"]},
                 ],
                 Owners=OS_TO_PCLUSTER_AMI_NAME_OWNER_MAP.get(os).get("owners"),
             ).get("Images", [])
-            return client.copy_image(
+            ami_id = client.copy_image(
                 Description="This AMI is a copy from an official AMI but uses a different naming. "
                 "It is used to bypass the AMI's name validation of pcluster version "
                 "to test the validation in Cookbook.",
                 Name=ami_name,
                 SourceImageId=official_amis[0]["ImageId"],
                 SourceRegion=region,
             ).get("ImageId")
+            logging.info("Sleeping 500 seconds for the newly copied AMI to become available")
+            time.sleep(500)
```

Comment:
Instead of waiting for a fixed amount of time I'd use a `describe-images` boto3 call with a retry and a timeout.

Reply:
You mean surround describe-images with a while loop until the state become available or timeout?

Reply:
Yes, you can create a function for this check and use the `@retry` annotation.
We already have the [retrying library](https://pypi.org/project/retrying/) in our code.

Something similar to:
```
@retry(
        stop_max_attempt_number=10,
        wait_fixed=30,
        retry_on_exception=lambda exception: isinstance(exception, ClientError),
    )
def retrieve_ami_id(...):
```